rama-net/src/tls/client/hello/mod.rs

```diff
@@ -234,8 +246,43 @@ impl ClientHelloExtension {
             }
             ClientHelloExtension::SupportedVersions(_) => ExtensionId::SUPPORTED_VERSIONS,
             ClientHelloExtension::CertificateCompression(_) => ExtensionId::COMPRESS_CERTIFICATE,
+            ClientHelloExtension::DelegatedCredentials(_) => ExtensionId::DELEGATED_CREDENTIAL,
             ClientHelloExtension::RecordSizeLimit(_) => ExtensionId::RECORD_SIZE_LIMIT,
+            ClientHelloExtension::EncryptedClientHello(_) => ExtensionId::ENCRYPTED_CLIENT_HELLO,
             ClientHelloExtension::Opaque { id, .. } => *id,
         }
     }
 }
+
+#[derive(Debug, Clone, Serialize, Deserialize, Hash)]
+/// Client Hello contents send by ech
+pub enum ECHClientHello {
+    /// Send when message is in the outer (unencrypted) part of client hello
+    Outer(ECHClientHelloOuter),
+    /// Send inside the inner (encrypted) client hello, need so all extensions
+    /// are part of the inner client hello. Servers will only use this hello
+    /// and if it does contain this extension the server thinks ech is not supported.
+    Inner(),
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, Hash)]
+/// Data send by ech hello message when it is in the outer part
+pub struct ECHClientHelloOuter {
+    pub cipher_suite: HpkeSymmetricCipherSuite,
+    pub config_id: u8,
+    pub enc: Vec<u8>,
+    pub payload: Vec<u8>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, Hash)]
+/// HPKE KDF and AEAD pair used to encrypt ClientHello
+pub struct HpkeSymmetricCipherSuite {
+    pub kdf_id: u16,
```

Comment:
According to <https://datatracker.ietf.org/doc/rfc9180/> these two idenfiiers have well defined values. Can you please add an enum for this to the file with other enums. It still allows for unknown values but at least gives a more verbose name for those that we do know about :)